bugwarrior/services/gitlab.py

```diff
@@ -500,71 +567,58 @@ def issues(self):
                 all_repos.append(item)
 
         else:
-            querystring = { 'simple': True }
+            querystring = {'simple': True}
             if self.config.membership:
                 querystring['membership'] = True
             if self.config.owned:
                 querystring['owned'] = True
             all_repos = self._fetch_paged(tmpl + '?' + urlencode(querystring))
+        return all_repos
 
-        repos = list(filter(self.filter_repos, all_repos))
-
-        repo_map = {}
-        issues = {}
-        for repo in repos:
-            rid = repo['id']
-            repo_map[rid] = repo
-            issues.update(
-                self.get_repo_issues(rid)
-            )
-        log.debug(" Found %i issues.", len(issues))
-        issues = list(filter(self.include, issues.values()))
-        log.debug(" Pruned down to %i issues.", len(issues))
-
-        yield from self._get_issue_objs(issues, 'issue', repo_map)
+    def issues(self):
+        if self.config.query:
+            item_type = self.get_type_from_query(self.config.query)
+            if item_type != "todo":
+                items = self.get_issues_from_query(self.config.query)
+
+                type_plural = item_type + 's'
+                log.debug("Found %i %s.", len(items), type_plural)
+                items = list(filter(self.include, items.values()))
+                log.debug("Pruned down to %i %s.", len(items), type_plural)
+                self.update_repo_map(items)
+                yield from self._get_issue_objs(items, item_type)
+            else:
+                yield from self.yield_todos_from_query('todos?state=pending')
```

Comment:
It seems like if the query starts with "todos" then anything else in the query will be ignored in favor of this hard coded query.

Reply:
yes, that's not intentional, thanks!